tests/webTables.spec.ts

```diff
@@ -148,7 +137,7 @@ test.describe("Web Tables-Specialties", async () => {
 
     const allBoxes = await page.locator(".dropdown-content div").allTextContents();
 
-    expect(allBoxes).toEqual(specialties);
+    await expect(allBoxes).toEqual(['radiology', 'surgery','dentistry', 'New specialty', 'oncology']);
```

Comment:
This is not a fix I was expecting. The initial version of the code was better than the hardcoded values of the specialties.

The goal of this test is to validate that the list of the specialties displayed on the Specialty page is the same as the list shown in the dropdown menu. For that, it's not needed to hardcode the expected specialties. You get the list from specialties page, saves to variable and compare it to the list in dropdown.

That this script is missing, is proper waiting for the list of specialties on the Sprcialty page to be available, before you save this list into "specialties" variable.
The issue happening, because method `specialtyRows.all()` method **do not wait** for all values to be displayed on the page after you add "Save" button adding "oncology" specialty. Because of that, you need to add additional wait before this step, making sure, all values in the specialties list are displayed before grabbing it from the page. Initially you had `waitForTimtout()` there, but I asked you to replace it with dynamic wait. Instead, you have deleted the wait completely that is caused the issue. 

You can read more in this article: https://www.bondaracademy.com/blog/playwright-not-waiting-for-elements

Please refactor your code to use a dynamic wait. You need to get this! :)